ibis/expr/types/joins.py

```diff
@@ -207,6 +209,9 @@ def select(self, *args, **kwargs):
         # if there are values referencing fields from the join chain constructed
         # so far, we need to replace them the fields from one of the join links
         subs = dereference_mapping_left(chain)
+        values = {
```

Comment:
I was experimenting with a more generic dereferencing method to allow inputs outside of the expression tree by searching the tree backwards until we find a field which can be dereferenced to the desired relation(s). It is promising, so we may be able to consolidate this.